draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -9717,9 +9717,33 @@ Content-Type: text/plain
 </section>
 </section>
 
-<section title="Field Name Extensibility" anchor="field-extensibility">
+<section title="Field Extensibility" anchor="field-extensibility">
   <x:anchor-alias value="header.extensibility"/>
 
+<t>
+  HTTP's most widely used extensibility point is the definition of new header and
+  trailer fields.
+</t>
+<t>
+  New fields can be defined such that, when they are understood by a
+  recipient, they override or enhance the interpretation of previously
+  defined fields, define preconditions on request evaluation, or
+  refine the meaning of responses.
+</t>
+<t>
+   However, by their nature HTTP fields cannot be required to be understood
+   by recipients without previous arrangement. For example, a new request
+   precondition cannot be assumed to be supported by servers; a server that
+   has not yet implemented it will ignore the field and process the request
+   as usual. Therefore, a function that uses new fields needs to either
+   degrade gracefully when handled by a recipient who does not understand
+   them, or use negotiation mechanism to establish whether the recipient
+   supports the fields in question. Typically, this will be out of band; for
+   example, using a particular media type might imply support for associated
+   header fields, or interacting with a defined well-known URI <xref
+   target="RFC8615"/> might do so.
```

Comment:
Other examples: response header fields (Accept-Ranges, Accept-CH), and OPTIONS.

Reply:
Could you say a bit more?

Reply:
Gotcha. So I happen to disagree with "typically out of band" because there are two IHMO widely used ways which are in band (OPTIONS and response header fields).

Reply:
Fair enough, will just list both in-band and out-of-band options...